### PR TITLE
Add jsdoc for some functions

### PR DESCRIPTION
**Pull request by ufko**
https://github.com/polltery/node-cli-game-example/issues/19
I tried to explain some functions and I'm completely open to rewriting them if you think they are not explanatory enough

### CODE REVIEW

File: src/map-engine.js [hunk 1]
```diff
@@ -83,6 +83,10 @@ mapEngine.displayPosition = function(player){
     console.log('you are at '+player.x+','+player.y);
 };
 
+/** Available tiles to which player
+ * can move from his current position
+ * @param {object} player - The player (Geralt of Rivia)
+ * */
```
Comment (polltery) [inline src/map-engine.js]:
```suggestion
* @returns {string []} An array containing options where the player is allowed to move to
* * /
```

File: src/map-engine.js [hunk 2]
```diff
@@ -109,6 +113,11 @@ mapEngine.getMovementOptions = function(player){
     return options;
 };
 
+/** If the player explored a tile, it's type is returned
+ *  which can be safe, blocked, monster, boss
+ *  If the player didn't explored a tile, 'unknown' is returned
+ *  @param {object} position - Player's next position
+ * */
```
Comment (polltery) [inline src/map-engine.js]:
```suggestion
 * @returns {string} 'unknown' | any other defined types such as 'special', 'monster', 'boss', 'friend'
 * */
```

File: src/map-engine.js [hunk 3]
```diff
@@ -153,6 +163,12 @@ mapEngine.displayMovementOptions = function(movementOptions, player){
     }
 };
 
+/** Returns the card (monster, boss, friend, special) in the tile
+ *  according to the player's current position in the map and the direction choice
+ *  @param {number} x - Player's position on the x axis.
+ *  @param {number} y - Player's position on the y axis.
+ *  @param {string} direction - Choice from user input.
+ * */
```
Comment (polltery) [inline src/map-engine.js]:
```suggestion
 * return {Object} returns the map tile from the specified direction.
 * */
```

**Comment by polltery** [comment]
Hey @ufko everything looks good, can you commit my suggestions so I can continue with the merge. Thanks!

**Comment by ufko** [comment]
Sure I'll make the changes now

**Pull request merged**